coderd/workspaces.go

```diff
@@ -1072,11 +1077,18 @@ func validWorkspaceSchedule(s *string, min time.Duration) (sql.NullString, error
 
 // workspaceSearchQuery takes a query string and returns the workspace filter.
 // It also can return the list of validation errors to return to the api.
-func workspaceSearchQuery(query string) (database.GetWorkspacesParams, []codersdk.ValidationError) {
+func workspaceSearchQuery(query string, page codersdk.Pagination) (database.GetWorkspacesParams, []codersdk.ValidationError) {
+	if page.Limit < 1 {
+		page.Limit = 65536
```

Comment:
why so big?

Reply:
I can't have it be a negative number, or 0, but I effectively don't want it to apply. This was 2^16 which seemed high enough to not practically get in the way.

Reply:
if the limit is 0 why not just omit it from the query?

Reply:
I wasn't sure how to do that exactly with the limit clause in postgres / sqlc - if you know of a way please share

Reply:
You can use a case statement
```
select 
	*
from
	organizations
limit case when @limit > 0 then @limit end;
```